src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs

```diff
@@ -12,18 +12,33 @@ public static class CacheExtensions
 
         public static object Get(this IMemoryCache cache, object key)
         {
+            if (key == null)
```

Comment:
Can we add a few tests to see that a ArgumentNullException is thrown. See: https://github.com/aspnet/Caching/blob/dev/test/Microsoft.Extensions.Caching.Redis.Tests/RedisCacheSetAndRemoveTests.cs#L93-L97 for a good example.

Reply:
I have added 6 test cases to test all the different scenario's and have also refactored the MemoryCacheExtensions.cs [CacheExtensions] to remove the redundant validations made in the earlier check-in. Please review and share your review comments.